http-ballerina/caching_response_age_calculation.bal

```diff
@@ -15,52 +15,53 @@
 // under the License.
 
 import ballerina/log;
-import ballerina/lang.'int;
+import ballerina/lang.'decimal;
 import ballerina/time;
 
 // Based on https://tools.ietf.org/html/rfc7234#section-4.2.3
 isolated function calculateCurrentResponseAge(Response cachedResponse) returns @tainted int {
-    int ageValue = getResponseAge(cachedResponse);
-    int dateValue = getDateValue(cachedResponse);
-    int now = time:currentTime().time;
-    int responseTime = cachedResponse.receivedTime;
-    int requestTime = cachedResponse.requestTime;
+    time:Seconds ageValue = getResponseAge(cachedResponse);
+    time:Utc dateValue = getDateValue(cachedResponse);
+    time:Utc now = time:utcNow();
+    time:Utc responseTime = cachedResponse.receivedTime;
+    time:Utc requestTime = cachedResponse.requestTime;
 
-    int apparentAge = (responseTime - dateValue) >= 0 ? (responseTime - dateValue) : 0;
+    time:Seconds ageDiff = time:utcDiffSeconds(responseTime, dateValue);
+    time:Seconds apparentAge = ageDiff >= 0 ? ageDiff : 0;
 
-    int responseDelay = responseTime - requestTime;
-    int correctedAgeValue = ageValue + responseDelay;
+    time:Seconds responseDelay = time:utcDiffSeconds(responseTime, requestTime);
+    time:Seconds correctedAgeValue = ageValue + responseDelay;
 
-    int correctedInitialAge = apparentAge > correctedAgeValue ? apparentAge : correctedAgeValue;
-    int residentTime = now - responseTime;
+    time:Seconds correctedInitialAge = apparentAge > correctedAgeValue ? apparentAge : correctedAgeValue;
+    time:Seconds residentTime = time:utcDiffSeconds(now, responseTime);
 
-    return (correctedInitialAge + residentTime) / 1000;
+    return <int>((correctedInitialAge + residentTime)*1000);
```

Comment:
No need for multiplying here. Also what's the need to replace int with `time:Seconds`? Since these are internal stuff I mean

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-http/pull/273